### PR TITLE
[B5 Migration] Upgrade Crispy Forms and reference our custom transition template for migration

### PR DESCRIPTION
**Pull request by biyeun**
## Technical Summary
This upgrades `django-crispy-forms` to version `2.0` and also installs our own `crispy-bootstrap3to5` template pack.
See [PR for the split here](https://github.com/dimagi/crispy-bootstrap3to5/pull/1).

### Safety story
This will definitely need QA before merge as it affects all forms on HQ.

### QA Plan
Please test:
- Project settings and My Settings forms
- SMS/messenging forms
- Accounting forms
- Create new account form
- Create new mobile worker form
- Create Web User form
- Any other form-based critical workflows

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -240,11 +241,8 @@
 RECAPTCHA_PRIVATE_KEY = ''
 RECAPTCHA_PUBLIC_KEY = ''
 
-CRISPY_TEMPLATE_PACK = 'bootstrap3'
-CRISPY_ALLOWED_TEMPLATE_PACKS = (
-    'bootstrap',
-    'bootstrap3',
-)
+CRISPY_TEMPLATE_PACK = 'bootstrap3to5'
+CRISPY_ALLOWED_TEMPLATE_PACKS = "bootstrap3to5"
```
Comment (millerdev) [inline settings.py:245]: Previously this was a tuple, now it's a string. Are both of those types supported for this setting?

**Review by millerdev — APPROVED** [review]

File: settings.py (hunk 1 above)
Comment (biyeun) [reply settings.py:245]: Great question, and something that confused me too. According [to the docs](https://github.com/django-crispy-forms/crispy-bootstrap5/#usage), this is now a string? There's no mention of `CRISPY_ALLOWED_TEMPLATE_PACKS`[ in the read the docs](https://django-crispy-forms.readthedocs.io/en/latest/install.html#template-packs), and the previous functionality to switch template packs that I vaguely remember seems to be completely removed.

File: requirements/dev-requirements.txt [hunk 2]
```diff
@@ -747,7 +754,7 @@ sphinxcontrib-serializinghtml==1.1.5
     # via sphinx
 sqlagg==0.17.2
     # via -r base-requirements.in
-sqlalchemy==1.3.19
```
Comment (esoergel) [inline requirements/dev-requirements.txt:750]: Looks like this is a breaking change.
`sqlagg` uses a method called `append_group_by` here:

https://github.com/dimagi/sql-agg/blob/98d20c177087272b57d525ab58facbe6b7f979f1/sqlagg/base.py#L164

As of sqlalchemy 1.4, this was deprecated:
https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_4_49/lib/sqlalchemy/sql/selectable.py#L3425

and it looks to have been removed in the 2.0 release line:
https://github.com/sqlalchemy/sqlalchemy/blob/rel_2_0_19/lib/sqlalchemy/sql/selectable.py

This causes a hard failure restoring users that have mobile UCRs configured: 

```
  <full trace omitted>
  File "/home/ethan/commcare-hq/corehq/apps/userreports/sql/data_source.py", line 64, in get_data
    ret = super(ConfigurableReportSqlDataSource, self).get_data(start=start, limit=limit)
  File "/home/ethan/commcare-hq/corehq/apps/reports/sqlreport.py", line 272, in get_data
    data = self._get_data(start=start, limit=limit)
  File "/home/ethan/commcare-hq/corehq/apps/reports/sqlreport.py", line 288, in _get_data
    return qc.resolve(session.connection(), self.filter_values)
  File "/home/ethan/.pyenv/versions/hq/lib/python3.9/site-packages/sqlagg/base.py", line 291, in resolve
    result = qm.execute(self.connection, filter_values or {})
  File "/home/ethan/.pyenv/versions/hq/lib/python3.9/site-packages/sqlagg/base.py", line 100, in execute
    query = self._build_query()
  File "/home/ethan/.pyenv/versions/hq/lib/python3.9/site-packages/sqlagg/base.py", line 139, in _build_query
    return self._build_query_generic(
  File "/home/ethan/.pyenv/versions/hq/lib/python3.9/site-packages/sqlagg/base.py", line 164, in _build_query_generic
    query.append_group_by(column(group_key))
AttributeError: 'Select' object has no attribute 'append_group_by'
```

Comment (esoergel) [reply requirements/dev-requirements.txt:750]: update: it looks like it also breaks if you attempt to view the UCR in HQ.  The UCR I'm testing with is a trivial one built using report builder - no aggregation, just copying properties over from the case

**Pull request merged**